Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -2410,6 +2425,28 @@ static IntRect computeMarqueeSelectionRect(const WebCore::IntPoint& point1, cons
     return true;
 }
 
+void UnifiedPDFPlugin::collectFindMatchRects(const String& target, WebCore::FindOptions options)
+{
+    m_findMatchRectsInDocumentCoordinates.clear();
+
+    auto foundSelections = [m_pdfDocument findString:target withOptions:compareOptionsForFindOptions(options)];
+    for (PDFSelection *selection in foundSelections) {
+        for (PDFPage *page in selection.pages) {
+            auto bounds = IntRect([selection boundsForPage:page]);
+            m_findMatchRectsInDocumentCoordinates.append(convertFromPageToDocument(bounds, *m_documentLayout.indexForPage(page)));
+        }
+    }
+
+    m_frame->page()->findController().didInvalidateFindRects();
```

Comment:
`protectedPage()` as it owns the FindController and you want to make sure it stays alive during the whole `didInvalidateFindRects()` execution.

Reply:
Sure!